From e1c4289d62ffd19ff25b5cfd499b84fde02cf524 Mon Sep 17 00:00:00 2001
From: fafryd <dz1125.bug.tracker@gmail.com>
Date: Thu, 10 Oct 2019 23:49:37 +0200
Subject: [PATCH] meson: Fix Xorg.wrap

/usr/bin/Xorg In xorg-server 1.20.0 is no longer a wrapper script. Instead it's a binary that appears to be id
entical to the one at /usr/lib/Xorg .
This breaks the execution of X with root rights.

Gitlab: https://gitlab.freedesktop.org/xorg/xserver/issues/539
---
 hw/xfree86/common/meson.build | 11 +++++++++++
 hw/xfree86/meson.build        |  8 +-------
 meson.build                   |  2 +-
 3 files changed, 13 insertions(+), 8 deletions(-)

diff --git a/hw/xfree86/common/meson.build b/hw/xfree86/common/meson.build
index a59aeb98f..1edf715c3 100644
--- a/hw/xfree86/common/meson.build
+++ b/hw/xfree86/common/meson.build
@@ -1,3 +1,14 @@
+if get_option('suid_wrapper')
+     configure_file(
+         input: '../Xorg.sh.in',
+         output: 'Xorg',
+         configuration: conf_data,
+         install: true,
+         install_dir: join_paths(get_option('prefix'), get_option('bindir')),
+         install_mode: 'rwxr-xr-x',
+     )
+endif
+
 srcs_xorg_common = [
     'xf86fbBus.c',
     'xf86noBus.c',
diff --git a/hw/xfree86/meson.build b/hw/xfree86/meson.build
index cf1f1ba78..9651de6f2 100644
--- a/hw/xfree86/meson.build
+++ b/hw/xfree86/meson.build
@@ -142,13 +142,7 @@ if get_option('suid_wrapper')
         c_args: xorg_c_args,
         install: true,
         install_dir: get_option('libexecdir'),
-        # install_mode: ['r-sr-xr-x', 0, 0],
-    )
-    configure_file(
-        input: 'Xorg.sh.in',
-        output: 'Xorg',
-        configuration: conf_data,
-        install_dir: join_paths(get_option('prefix'), get_option('bindir')),
+        install_mode: 'rwsr-xr-x',
     )
 endif
 
diff --git a/meson.build b/meson.build
index 2eca5c295..1961206ed 100644
--- a/meson.build
+++ b/meson.build
@@ -4,7 +4,7 @@ project('xserver', 'c',
             'c_std=gnu99',
         ],
         version: '1.20.99.1',
-        meson_version: '>= 0.46.0',
+        meson_version: '>= 0.50.0',
 )
 add_project_arguments('-DHAVE_DIX_CONFIG_H', language: ['c', 'objc'])
 cc = meson.get_compiler('c')
-- 
2.23.0

